bindings/wallet-jni/src/lib.rs

```diff
@@ -123,3 +121,101 @@ pub unsafe extern "system" fn Java_com_iohk_jormungandrwallet_Wallet_initialFund
     }
     settings as jlong
 }
+
+#[no_mangle]
+pub unsafe extern "system" fn Java_com_iohk_jormungandrwallet_Wallet_convert(
+    env: JNIEnv,
+    _: JClass,
+    wallet: jlong,
+    settings: jlong,
+) -> jlong {
+    let wallet_ptr = wallet as WalletPtr;
+    let settings_ptr = settings as SettingsPtr;
+    let mut conversion_out = null_mut();
+    let result = wallet_convert(
+        wallet_ptr,
+        settings_ptr,
+        (&mut conversion_out) as *mut ConversionPtr,
+    );
+
+    if let Some(error) = result.error() {
+        env.throw(error.to_string());
+    }
+
+    conversion_out as jlong
+}
+
+#[no_mangle]
+pub unsafe extern "system" fn Java_com_iohk_jormungandrwallet_Conversion_transactionsSize(
+    _env: JNIEnv,
+    _: JClass,
+    conversion: ConversionPtr,
+) -> jint {
+    let conversion = conversion as ConversionPtr;
+    // TODO: i don't think this can actually overflow, but probably best to have someone confirm that too
+    wallet_convert_transactions_size(conversion) as i32
+}
+
+#[no_mangle]
+pub unsafe extern "system" fn Java_com_iohk_jormungandrwallet_Conversion_transactionsGet(
+    env: JNIEnv,
+    _: JClass,
+    conversion: ConversionPtr,
+    index: jint,
+) -> jbyteArray {
+    let conversion = conversion as ConversionPtr;
+    // TODO: check here that index is non-negative? or maybe just do it in java code...
+    let index = index as usize;
+    let mut transaction_out: *const u8 = null();
+    let mut transaction_size: usize = 0; // XXX: unitialized?
+
+    let result = wallet_convert_transactions_get(
+        conversion,
+        index,
+        (&mut transaction_out) as *mut *const u8,
+        (&mut transaction_size) as *mut usize,
+    );
+
+    // TODO: maybe we can use a ByteBuffer or something here to avoid the copy
+    // (especially considering that there is already a requirement to call conversion's delete)
+    let array = env.new_byte_array(transaction_size as jint).expect("");
+    match result.error() {
```

Comment:
Oh wow, this is weird, so we are mattching the result but still creates a warning? 🙉

Reply:
I think it's only about the `env.throw`